core/src/test/java/org/apache/iceberg/TestMetricsConfig.java

```diff
@@ -28,18 +29,47 @@
 public class TestMetricsConfig {
 
   @Test
-  public void testNestedStructsRespectedInLimit() {
+  public void testNestedStruct() {
     Schema schema =
         new Schema(
             required(
                 1,
-                "col_struct",
+                "top_struct_a1",
                 Types.StructType.of(
-                    required(2, "a", Types.IntegerType.get()),
-                    required(3, "b", Types.IntegerType.get()))),
-            required(4, "top", Types.IntegerType.get()));
+                    required(2, "middle_b1", Types.IntegerType.get()),
+                    required(
+                        3,
+                        "middle_struct_b1",
+                        Types.StructType.of(optional(4, "bottom_c1", Types.StringType.get()))))),
```

Comment:
we should also add a primitive field after this struct to illustrate the traverse order